### PR TITLE
Feature/create earned points for employee

### PR DESCRIPTION
**Pull request by martinosso**

### CODE REVIEW

File: app/models/employee.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ class Employee < ApplicationRecord
   validates :number_of_available_kudos, presence: true
   has_many :given_kudos, class_name: 'Kudo', foreign_key: 'giver_id', inverse_of: 'giver', dependent: :destroy
   has_many :received_kudos, class_name: 'Kudo', foreign_key: 'receiver_id', inverse_of: 'receiver', dependent: :destroy
+  composed_of :earned_points, class_name: 'Money', mapping: %w[earned_points cents],
+                              converter: proc { |value| Money.new(value) }
```
Comment (rniedziolka) [inline app/models/employee.rb:8]: Generally, I can see it works, however, I have a hint that might make you work less the next time.
Are your `:earned_points` and `:received_kudos` the same?

**Review by rniedziolka — APPROVED** [review]
Hi martinosso,
Nice to e-meet you! 😄
Nice one 🚀
I left one hint.

**Review by ciejj — APPROVED** [review]
Interesting approach, but why have you not simply use number of received kudos as number of points? 
Your approach is meeting all the Acceptance Criteria, but is more complex. Also what happens to the employees that are already in db and have some received kudos? How many point will they have?

**Comment by martinosso** [comment]
> Interesting approach, but why have you not simply use number of received kudos as number of points? Your approach is meeting all the Acceptance Criteria, but is more complex. Also what happens to the employees that are already in db and have some received kudos? How many point will they have?

I know that using money gem on earned points and rewards is not required but it was in learning materials in spring 3 task 3 so i decided to try to use it. I forgot to set earned points on migration, I've already added it in last commit.

**Pull request merged**